### PR TITLE
Add computed relationships

### PR DESCRIPTION
**Pull request by steve-chavez**
Closes https://github.com/PostgREST/postgrest/issues/2144

## TODO

- [x] Override detected relationships
- [x] Tests
  - [x] o2m and m2o relationship
  - [x] rpc
  - [x] mutations
  - [x] override detected relationships
  - [x] self relationships
- [x] Refactor
- [x] CHANGELOG

### CODE REVIEW

**Comment by steve-chavez** [comment]
Should be ready. Overall I feel confident that this one doesn't break existing functionality since `ComputedRelationShip` is added as a new data constructor of `Relationship`.

File: src/PostgREST/DbStructure/Relationship.hs [hunk 1]
```diff
@@ -26,6 +26,13 @@ data Relationship = Relationship
   , relTableIsView  :: Bool
   , relFTableIsView :: Bool
   }
+  | ComputedRelationShip
```
Comment (laurenceisla) [inline src/PostgREST/DbStructure/Relationship.hs:29]:
```suggestion
  | ComputedRelationship
```

File: src/PostgREST/DbStructure.hs [hunk 2]
```diff
@@ -643,6 +662,46 @@ allM2ORels pgVer =
...
+      where relkind in ('v','r','m','f','p')
+    ),
+    computed_rels as (
+      select
+        p.pronamespace::regnamespace::text as schema,
+        p.proname as name,
+        split_part((p.proargtypes[0])::regtype::text, '.', 1) as rel_table_schema,
+        split_part((p.proargtypes[0])::regtype::text, '.', 2) as rel_table_name,
+        split_part(p.prorettype::regtype::text, '.', 1) as rel_ftable_schema,
+        split_part(p.prorettype::regtype::text, '.', 2) as rel_ftable_name,
```
Comment (laurenceisla) [inline src/PostgREST/DbStructure.hs:683]: It looks like this will break when a table has a `.` in its name (probably it's the same with the schema name).

Comment (steve-chavez) [reply src/PostgREST/DbStructure.hs]: Good catch. Fixed.

**Review by laurenceisla — APPROVED** [review]

**Pull request merged**